src/main/java/org/ecommerce/controllers/AdminController.java

```diff
@@ -4,12 +4,11 @@
 import org.ecommerce.models.Response;
 
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
 
 public class AdminController implements ControllerOperations<Admin, Long> {
     @Override
-    public Response<Admin> create(Map<String, String> request) {
+    public Response<Admin> create(String jsonRequest) {
```

Comment:
Changes are ok. Although, the less you update the signature of a public method, the better.

Reply:
Please create `Request` class and replace `String jsonRequest `.

Reply:
I actually did some kind of implementation for the requests. But I didn't include them in the PR because I wasn't sure if it was ok to refactor the controllers with a request model. but now that you mentioned it, here is the branch: https://github.com/algutierrez-griddynamics/e-commerce/commit/5ef12aabcca2653225d45f1438b883f97e0b4925. should I merge it into the PR? 👀